### PR TITLE
Experimentation API pay gate

### PR DESCRIPTION
**Pull request by paolodamico**
## Changes
- Adds a pay gate to the Experimentation API too. First of its kind.

## How did you test this code?
- Django functional test.

### CODE REVIEW

**Comment by paolodamico** [comment]
@neilkakkar if you have Experimentation E2E tests, we might need to update this so there's a premium license available too.

File: posthog/api/async_migration.py [hunk 1]
```diff
@@ -50,7 +49,7 @@ class Meta:
 
 class AsyncMigrationsViewset(StructuredViewSetMixin, viewsets.ModelViewSet):
     queryset = AsyncMigration.objects.all()
-    permission_classes = [StaffUser]
+    permission_classes = [permissions.IsAuthenticated, permissions.IsAdminUser]
```
Comment (paolodamico) [inline posthog/api/async_migration.py:52]: Unrelated cleanup, there's a DRF global staff user permission.

File: posthog/constants.py [hunk 2]
```diff
@@ -15,7 +15,8 @@ class AvailableFeature(str, Enum):
     PATHS_ADVANCED = "paths_advanced"
     CORRELATION_ANALYSIS = "correlation_analysis"
     GROUP_ANALYTICS = "group_analytics"
-    MULTIVARIATE_FLAGS = "multivariate_flags"
+    MULTIVARIATE_FLAGS = ("multivariate_flags",)
```
Comment (neilkakkar) [inline posthog/constants.py:18]: I'm guessing you didn't actually want to add a comma (which I expect got turned into this via lint)

File: posthog/api/async_migration.py (hunk 1 above)
Comment (neilkakkar) [reply posthog/api/async_migration.py:52]: This is borking some tests^ (and some others are failing, related to project & enterprise permissions 👀 )

Comment (paolodamico) [reply posthog/api/async_migration.py]: fixed

File: posthog/constants.py (hunk 2 above)
Comment (paolodamico) [reply posthog/constants.py]: good catch, fixed

**Review by neilkakkar — APPROVED** [review]
looks good!

**Comment by neilkakkar** [comment]
Failures seem unrelated, re running 👀

**Comment by EDsCODE** [comment]
snapshots need to be updated. The person distinct id was updated but the snapshots everywhere weren't
